xpc-connection/tests/message_round_trip.rs

```diff
@@ -0,0 +1,178 @@
+use futures::executor::block_on;
+use futures::StreamExt;
+use std::collections::HashMap;
+use xpc_connection::{Message, XpcConnection};
+
+/// FIXME: Outgoing strings, both as dictionary keys and as string values, are
+/// expected to be explicitly NULL terminated, but incoming strings do not
+/// contain the NULL. This prevents us from being able to compare outgoing and
+/// incoming container dictionaries directly.
+/// This will also make writing an XPC listener a little more confusing as
+/// messages have two different behaviours depending on the context.
```

Comment:
Should we perhaps only accept `CString` instead of `String`?

Reply:
I think so, but it will be a breaking change to the API so I don't want to do it as part of this.